content/blog/its-time-to-try-out-pixi/contents.lr

```diff
@@ -0,0 +1,995 @@
+Ideally, this would be done with one command, e.g., `pytest`, with minimal overhead.
+The software testing environment should be the same as my development environment.
+
+### Both personas' needs
+
+#### Simplified environment specification
+
+As mentioned above, I currently use `environment.yml` and `pyproject.toml`
+to specify runtime and development dependencies.
+Runtime dependencies are declared in `pyproject.toml`,
+while development dependencies are declared in `environment.yml`.
+However, this way of separating concerns means that
+we have duplication in our dependencies specification!
+Ideally, we'd like to do this with a single file.
+
+#### Tooling for fast environment builds
+
+Whether we are in a Docker container or not,
+I'd like to see much faster container and environment build times
+with tooling for caching when compared to what I currently experience.
+
+(Note to self:
+Product-oriented folks keep talking about "don't solution, tell me the problem",
+but I have to say -- I only really knew how much of a problem this was
+once I touched the solution I'm about to talk about!)
+
+#### Automatic updates of lock files
+
+When I add a dependency to my environment,
+I'd like to see the lock file automatically updated
+so that I don't have to remember to do that in the future.
+
+## How pixi operates
+
+With the desiderata outlined above as contextual information,
+let's now see how `pixi` works.
+
+### Installation
+
+What's awesome is that `pixi` is installed via a simple, one-bash-script install.
+Official instructions are available [here](https://pixi.sh/latest/#installation),
+but as of the time of writing, the only thing we need to run is:
+
+```bash
+curl -fsSL https://pixi.sh/install.sh | bash
+```
+
+This sets up `pixi` within your home directory
+and adds the `pixi` path to your [PATH environment variable](https://ericmjl.github.io/essays-on-data-science/software-skills/environment-variables/).
+
+I also recommend getting set up with autocompletion for your shell.
+Once again, official instructions are [here](https://pixi.sh/latest/#autocompletion),
+and what I used for the `zsh` (macOS' default shell) was:
+
+```bash
+echo 'eval "$(pixi completion --shell zsh)"' >> ~/.zshrc
+```
+
+### Starting a new project with `pixi` enabled
+
+For those who are used to `conda` idioms,
+`pixi` won't create a centralized `conda` environment
+but instead will create a `.pixi` sub-directory
+within the directory of your `pixi` configuration file
+(either `pyproject.toml` or `pixi.toml`).
+In line with my desire to be able to specify sets of dependencies
+for runtime environments and development environments
+and also minimize the number of configuration files present,
+it makes sense to initialize with a `project.toml` configuration
+rather than the default `pixi.toml` configuration file.
+This is done by executing:
+
+```bash
+pixi init --pyproject -c conda-forge -v
```

Comment:
To not get a deprecation warning in the latest release (of today) the new cli would be:
```suggestion
pixi init --format pyproject -c conda-forge -v
```
Also you could leave out `-c conda-forge` as we default to that, if you would like that.
```suggestion
pixi init --format pyproject  -v
```